mismi-openssl/src/Mismi/OpenSSL/Control.hs

```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE NoImplicitPrelude  #-}
+{-# LANGUAGE LambdaCase         #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE RankNTypes #-}
+module Mismi.OpenSSL.Control (
+    A.AWS
+  , A.Error
+  , A.AccessKey
+  , A.SecretKey
+  , A.SessionToken
+  , A.Region (..)
+  , X.withOpenSSL
+  , runAWS
+  , runAWST
+  , runAWSTWith
+  , runAWSTWithRegion
+  , C.rawRunAWS
+  , runAWSWithRegion
+  , awsBracket
+  , awsBracket_
+  , C.newEnvFromCreds
+  , C.renderError
+  , C.setServiceRetry
+  , C.setRetry
+  , C.configureRetries
+  ) where
+
+import           Control.Lens (over)
+import           Control.Monad.Catch
+import           Control.Monad.IO.Class
+import           Control.Monad.Reader
+
+import qualified Mismi.Control as C
+import           Mismi.OpenSSL.Environment
+
+import           Network.AWS hiding (runAWS)
+import qualified Network.AWS as A
+
+import           Network.HTTP.Client.Internal (mResponseTimeout)
+import qualified Network.HTTP.Client.OpenSSL as X
+import           Network.HTTP.Client.OpenSSL (withOpenSSL)
+
+import           P
+
+import           System.IO
+
+import           X.Control.Monad.Trans.Either
+
+runAWST :: Env -> (Error -> e) -> EitherT e AWS a -> EitherT e IO a
+runAWST e err action =
+  runAWSTWith (runAWS e) err action
+
+runAWSTWithRegion :: Region -> (Error -> e) -> EitherT e AWS a -> EitherT e IO a
+runAWSTWithRegion r err action =
+  runAWSTWith (runAWSWithRegion r) err action
+
+runAWSTWith :: (forall b. AWS b -> EitherT Error IO b) -> (Error -> e) -> EitherT e AWS a -> EitherT e IO a
+runAWSTWith run err action =
+  joinErrors id err $ mapEitherT run action
+
+runAWS :: (MonadIO m, MonadCatch m) => Env -> AWS a -> EitherT Error m a
+runAWS e'' =
+  let
+    e' = over envManager (\m -> m { mResponseTimeout = Just 60000000 }) e''
+    e = C.configureRetries 5 e'
+  in
+    EitherT . try . liftIO . withOpenSSL . C.rawRunAWS e
+
+runAWSWithRegion :: (MonadIO m, MonadCatch m) => Region -> AWS a -> EitherT Error m a
+runAWSWithRegion r a = do
+  e <- liftIO $ discoverAWSEnvWithRegion r
+  runAWS e a
+
+awsBracket :: AWS a -> (a -> AWS c) -> (a -> AWS b) -> AWS b
+awsBracket r f a = do
+  e <- ask
+  liftIO $ bracket (unsafeRunAWS e r) (unsafeRunAWS e . f) (unsafeRunAWS e . a)
+
+awsBracket_ :: AWS a -> AWS c -> AWS b -> AWS b
+awsBracket_ r f a =
+  awsBracket r (const f) (const a)
+
+unsafeRunAWS :: Env -> AWS a -> IO a
+unsafeRunAWS e a =
```

Comment:
That is different and re-exported, this is built on top of our runAWS and only used in the awsBracket